thrimshim/thrimshim/main.py

```diff
@@ -242,22 +242,25 @@ def get_row(ident):
 		start = response['event_start']
 
 		# use tags to determine default thumbnail template
-		if response['thumbnail_template'] is None
+		if response['thumbnail_template'] is None:
 			conn = app.db_manager.get_conn()
 			query = """
-				SELECT name, default_template
+				SELECT tags, default_template
 				FROM playlists
-					"""
+				WHERE default_template IS NOT NULL
+			"""
 			results = database.query(conn, query)
 			default_templates = {}
 			for row in results:
-				row = row._asdict:
-				if row['name'] and row['default_template']:
-					default_templates[row['name']] = row['default_template']
+				for tag in row.tag:
+					default_templates[tag] = row.default_template
```

Comment:
I don't think this is correct. Suppose you have a playlist with tags `foo` and `bar`. This code will use that playlist's default template if `foo` *or* `bar` is in the event tags, whereas the playlist only matches if `foo` *and* `bar` are in the event tags.
It would also break if there is one playlist that matches `foo` and another that matches `foo` and `bar`, ie. it assumes tags are unique.

The multi-tag stuff is confusing, I think the easiest option for now is to only consider single-tag playlists. But if we do  support multiple tags we should only match if the video would be added to the playlist. It's just a question of how that would interact with the tag order priority. Maybe take the matching playlist with the latest tag that is part of that playlist's tags? so eg. with tags `foo, bar, baz` on the event, and playlists `foo, baz` and `bar`, `foo, baz` would take priority because `baz` comes last.